spec/unit/provider/package/gem_spec.rb

```diff
@@ -36,6 +40,34 @@
         allow(provider).to receive(:rubygem_version).and_return('1.9.9')
       end
 
+      context 'on windows' do
+        let(:path) do
+          "C:\\Program Files\\Puppet Labs\\Puppet\\puppet\\bin;C:\\Program Files\\Puppet Labs\\Puppet\\bin;C:\\Ruby26-x64\\bin;C:\\Windows\\system32\\bin"
+        end
+
+        let(:expected_path) do
+          "C:\\Program Files\\Puppet Labs\\Puppet\\bin;C:\\Ruby26-x64\\bin;C:\\Windows\\system32\\bin"
+        end
+
+        before do
+          allow(Puppet::Util::Platform).to receive(:windows?).and_return(true)
+          allow(Puppet::Util).to receive(:get_env)
+          allow(Puppet::Util).to receive(:get_env).with('PATH').and_return(path)
```

Comment:
puppet's spec_helper restores the environment across tests (https://github.com/puppetlabs/puppet/blob/b3fe419f5170d8726f493d707b3751ccbf7c1bbd/lib/puppet/test/test_helper.rb#L194-L201) so you could avoid stubbing and do:

```
ENV['PATH'] = "#{path};#{ENV['PATH']}"
```

But then we're mucking with the running ruby process' PATH...

Reply:
I think is better contained if we mock only the part that we are interested in instead of the `PATH` env. I wouldn't expect that mocking `get_env('PATH')` to interfere with other parts of puppet and make this test unreliable.